api/src/main/java/org/hawkular/inventory/api/filters/Filter.java

```diff
@@ -42,9 +42,10 @@
 public abstract class Filter {
     private static final Filter[] EMPTY = new Filter[0];
 
-    Filter() {
-
-    }
+//    we should not prohibit creating sub-classes of Filter in the implementations of this API
```

Comment:
I needed to add the `JumpInOutFilter` for "jumping" to the edges or from the edges during the graph traversal. At the same time the filter should be an implementation detail in the library implementation (default visibility).

> True, but on the other hand, this enables the user of the library to do the same and expect the implementation would work with it.

JDoc? Right now the `FilterApplicator` kind of serves as a white list of allowed filters. If the library consumer naively extends a filter with the hope it'll work. It fails with the first usage of the filter with

`IllegalArgumentException("Unsupported filter type " + filter.getClass());`

I'd argue that using the library != extending it's classes and pass this derived classes naively to the library and expects it'll work. It's not a big deal if we comment it in JDoc, I think.

The "in-memory" application sound interesting. Do you mean applying the filters on the network of interconnected Java objects or using something like ispn?